src/eval.rs

```diff
@@ -1,16 +1,18 @@
-use types::{Expression, EvaluationError, EvalResult, SpecialForm, Atom, Environment, FunctionType, SinglyLinkedList};
-use nom::lib::std::collections::{HashMap};
+use types::{Expression, EvaluationError, EvalResult, SpecialForm, Atom, Environment, FunctionType, SinglyLinkedList, MapType};
+use nom::lib::std::collections::{HashMap, VecDeque};
+use util;
 
-fn quote_expr(to_quote: Vec<Expression>) -> EvalResult<Expression> {
+pub type LocalVars<'a> = SinglyLinkedList<'a, HashMap<String, Expression>>;
+
+fn quote_expr(to_quote: VecDeque<Expression>) -> EvalResult<Expression> {
     if to_quote.len() == 1 {
         Ok(to_quote[0].clone())
     } else {
         Err(EvaluationError::WrongArity(1, to_quote.len() as i64))
     }
-
 }
 
-fn if_expr(args: Vec<Expression>, env: &mut Environment, local_vars: &SinglyLinkedList<HashMap<String, Expression>>) -> EvalResult<Expression> {
+fn if_expr(args: VecDeque<Expression>, env: &mut Environment, local_vars: &LocalVars) -> EvalResult<Expression> {
```

Comment:
`VecDeque` is more efficient to add to front/drop c: now `List` more closely resembles an actual linked list